src/bacnet/basic/object/time_value.c

```diff
@@ -0,0 +1,694 @@
+ *
+ * @param  rpdata - BACNET_READ_PROPERTY_DATA data, including
+ * requested data and space for the reply, or error response.
+ *
+ * @return number of APDU bytes in the response, or
+ * BACNET_STATUS_ERROR on error.
+ */
+int Time_Value_Read_Property(BACNET_READ_PROPERTY_DATA *rpdata)
+{
+    int apdu_len = 0; /* return value */
+    BACNET_BIT_STRING bit_string;
+    BACNET_CHARACTER_STRING char_string;
+    uint8_t *apdu = NULL;
+    BACNET_TIME value;
+
+    if ((rpdata == NULL) || (rpdata->application_data == NULL) ||
+        (rpdata->application_data_len == 0)) {
+        return 0;
+    }
+
+    apdu = rpdata->application_data;
+    switch (rpdata->object_property) {
+        case PROP_OBJECT_IDENTIFIER:
+            apdu_len = encode_application_object_id(
+                &apdu[0], rpdata->object_type, rpdata->object_instance);
+            break;
+        case PROP_OBJECT_NAME:
+            Time_Value_Object_Name(rpdata->object_instance, &char_string);
+            apdu_len =
+                encode_application_character_string(&apdu[0], &char_string);
+            break;
+        case PROP_OBJECT_TYPE:
+            apdu_len =
+                encode_application_enumerated(&apdu[0], rpdata->object_type);
+            break;
+        case PROP_PRESENT_VALUE:
+            if (Time_Value_Present_Value(rpdata->object_instance, &value)) {
+                apdu_len = encode_application_time(apdu, &value);
+            }
+            break;
+
+        case PROP_STATUS_FLAGS:
+            bitstring_init(&bit_string);
+            bitstring_set_bit(&bit_string, STATUS_FLAG_IN_ALARM, false);
+            bitstring_set_bit(&bit_string, STATUS_FLAG_FAULT, false);
+            bitstring_set_bit(&bit_string, STATUS_FLAG_OVERRIDDEN, false);
+            bitstring_set_bit(&bit_string, STATUS_FLAG_OUT_OF_SERVICE,
+                Time_Value_Out_Of_Service(rpdata->object_instance));
+            apdu_len = encode_application_bitstring(&apdu[0], &bit_string);
+            break;
+        case PROP_OUT_OF_SERVICE:
+            apdu_len = encode_application_boolean(&apdu[0],
+                Time_Value_Out_Of_Service(rpdata->object_instance));
+            break;
+        case PROP_DESCRIPTION:
+            characterstring_init_ansi(
+                &char_string, Time_Value_Description(rpdata->object_instance));
+            apdu_len = encode_application_character_string(apdu, &char_string);
+            break;
+        case PROP_EVENT_STATE:
+            apdu_len =
+                encode_application_enumerated(&apdu[0], EVENT_STATE_NORMAL);
+            break;
+        default:
+            rpdata->error_class = ERROR_CLASS_PROPERTY;
+            rpdata->error_code = ERROR_CODE_UNKNOWN_PROPERTY;
+            apdu_len = BACNET_STATUS_ERROR;
+            break;
+    }
+    /*  only array properties can have array options */
+    if ((apdu_len >= 0) && (rpdata->object_property != PROP_PRIORITY_ARRAY) &&
```

Comment:
Time Value object properties that are a BACnetARRAY: 

- Property_List BACnetARRAY[N] of BACnetPropertyIdentifier
- Event_Time_Stamps BACnetARRAY[3] of BACnetTimeStamp
- Event_Message_Texts BACnetARRAY[3] of CharacterString
- Event_Message_Texts_Config BACnetARRAY[3] of CharacterString
- Value_Source_Array BACnetARRAY[16] of BACnetValueSource
- Command_Time_Array BACnetARRAY[16] of BACnetTimeStamp
- Tags BACnetARRAY[N] of BACnetNameValue

Please add these properties to a function:
```
bool Time_Value_BACnetARRAY_Property(BACNET_PROPERTY_ID object_property)
{
    /* return true if the property is an BACnetArray */
}
```
It can be used by ReadProperty and WriteProperty functions.